instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpServerAttributesGetter.java

```diff
@@ -20,70 +20,16 @@
   // Attributes that always exist in a request
 
   @Nullable
-  default String getFlavor(REQUEST request) {
-    return flavor(request);
-  }
-
-  /**
-   * This method is deprecated and will be removed in the subsequent release.
-   *
-   * @deprecated Use {@link #getFlavor(Object)} instead.
-   */
-  @Deprecated
-  @Nullable
-  default String flavor(REQUEST request) {
-    throw new UnsupportedOperationException(
-        "This method is deprecated and will be removed in the subsequent release.");
-  }
+  String getFlavor(REQUEST request);
 
   @Nullable
-  default String getTarget(REQUEST request) {
-    return target(request);
-  }
-
-  /**
-   * This method is deprecated and will be removed in the subsequent release.
-   *
-   * @deprecated Use {@link #getTarget(Object)} instead.
-   */
-  @Deprecated
-  @Nullable
-  default String target(REQUEST request) {
-    throw new UnsupportedOperationException(
-        "This method is deprecated and will be removed in the subsequent release.");
-  }
+  String getTarget(REQUEST request);
 
   @Nullable
   default String getRoute(REQUEST request) {
-    return route(request);
+    return null;
```

Comment:
I made this return `null` by default because there is only one instrumentation (armeria) that actually implements that; most set the route using the `HttpRouteHolder` class